tests/bcl-test/BCLTests/templates/common/IgnoreFileParser.cs

```diff
@@ -0,0 +1,36 @@
+using System.IO;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace BCLTests {
+
+	/// <summary>
+	/// Class that can parser a file/stream with the ignored tests and will
+	/// return a list of the ignored tests.
+	/// </summary>
+	public static class IgnoreFileParser {
+
+		public static async Task<string[]> ParseStream (TextReader textReader)
```

Comment:
can't you use a `IEnumerable<strig>` here ?
that way you can return the `List<string>` itself, without a (allocating) call to `ToArray`